### ISSUE TITLE
Use UFO point labels as source for axes

### ISSUE THREAD

**Issue by djrrb** [TypeNetwork/Parama-roundup#11]
I’m thinking of using labels `com.typenetwork.parama-roundup.XXXX` where `XXXX` is the axis name.

We can use these to fill the data on the input sheet, rather than entering it manually.

**Comment by djrrb** [comment]
I’m working on implementing the point labels implementation of parama roundup. So the imagined workflow is this:

1. Insert measurements as labels in UFOs
2. Run a script to find glyphs and point indexes and write them to existing `Inputs` sheet
3. Run a script to parse measurements from `Inputs` sheet and write them to `Measurements` and `Widths` sheets

The point labels are stored in a custom RoboFont lib entry. They can be written and read by any software, but are not part of the UFO spec or currently implemented in non-Robofont environments AFAIK.  

@dberlow Does all of this sound good? Also, how do you want to handle point labels for phantom points?

**Comment by dberlow** [comment]
The origin point should be one index larger than the last point of the last contour of the glyph and the width poont should be two index points larger than the last point of the cliff as per the truetype spec.

**Comment by djrrb** [comment]
Yes, but the question is how do you want to use point labels on points that
don’t exist? 🙃

**Comment by dberlow** [comment]
The points and goemetry that got you to The phantom point  is the id?

**Comment by djrrb** [comment]
Yeah, before we were just typing ID numbers to a spreadsheet so it was easy to just type in slightly-bigger numbers for the phantom points. Now we are looking to label points directly in the UFO, which requires that a point exists in order to be labeled. 

Possible options: 

- Add anchors for phantom points (but then they have to be added to all masters)
- Write a script that labels phantom points as a custom lib entry (would not be visible in robofont)
- Hard code results with phantom points into the spreadsheet, bypassing labels altogether for those axes

**Comment by dberlow** [comment]
Curious, when RF generates a ttf, does it add the phantom points and then
throw them away when it reads them?

RF, UFO has to store the width (rsb) somewhere?, and technically the
location of the origin (lsb) is known to all points;) which is why
alignments are a single value. . .
so... can't we find values for calculations even without adding them?

**Comment by djrrb** [comment]
> Curious, when RF generates a ttf, does it add the phantom points and then
> throw them away when it reads them?

AFAIK there are no phantom points actually added into the TTF file...those index are just understood and can be referenced in the hints. The glyph width is stored in the `<hmtx>` and the y valuesare stored in the `<hhea>` and `<os_2>`.

> so... can't we find values for calculations even without adding them?

Yes we can calculate, but we can’t label at least in the same sense that we can label points. I’ll try to come up with a workaround.

**Comment by dberlow** [comment]
That is correcter, the scaler creates them from the hmtx table.

Can we label a value in the hmtx table as/if it were a point?

**Comment by djrrb** [comment]
Not using the RoboFont interface, unfortunately. And that was kind of the whole benefit of the labels, to make the process more intuitive.

**Comment by dberlow** [comment]
Just forget it then. I tried 10 years ago to get phantom points added to RF so we could hint there, I failed to convince our developer or to get them to convince Frederik to do so. I guess RF is now not up-To-date with the needs of people making TT. 

On Nov 18, 2019, at 4:50 PM, David Jonathan Ross <notifications@github.com> wrote:

Not using the RoboFont interface, unfortunately. And that was kind of the whole benefit of the labels, to make the process more intuitive.
